### DISCUSSION TITLE
How can we integrate UI libraries in the reactpy project?

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [idom-team/idom#1215] [General]
I like to use Chakra UI, ShadCN UI and Tailwind UI for the front-end project. How can I integrate it into this ReactPy?

**Comment by kerighan** [comment]
Any idea on how to achieve this?

**Reply by itsmeadarsh2008** [reply]
I found an alternative called Reflex.

**Comment by fsmosca** [comment]
I have a [dedicated blog for reactpy](https://energybeam.blogspot.com/). I use bootstrap.

**Comment by Archmonger** [comment]
By using `reactpy-material` as an example, you need to:

1. Create a JavaScript file that can [exposes the `bind` function](https://github.com/williamneto/reactpy-material/blob/8a165a39fa9b68ad59cd3237b1d68a4736476b76/js/src/index.js#L46-L55), and [exports the component](https://github.com/williamneto/reactpy-material/blob/8a165a39fa9b68ad59cd3237b1d68a4736476b76/js/src/index.js#L57-L59) you need from your library.
2. Bring those [components into python](https://github.com/williamneto/reactpy-material/blob/8a165a39fa9b68ad59cd3237b1d68a4736476b76/reactpy_material/core/components.py#L16) using `reactpy.web.module.export`
3. That's all! Now you can [call those components](https://github.com/williamneto/reactpy-material/blob/8a165a39fa9b68ad59cd3237b1d68a4736476b76/reactpy_material/core/components.py#L27-L29) within your ReactPy application.
